src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -483,10 +483,16 @@ private async Task<bool> AreCentralVersionRequirementsSatisfiedAsync(RestoreRequ
                 return false;
             }
 
-            // Log a warning if there are more than one configured source and package source mapping is not enabled
-            if (restoreRequest.Project.RestoreMetadata.Sources.Count > 1 && !restoreRequest.PackageSourceMapping.IsEnabled)
+            if (!restoreRequest.PackageSourceMapping.IsEnabled)
             {
-                await _logger.LogAsync(RestoreLogMessage.CreateWarning(NuGetLogCode.NU1507, string.Format(CultureInfo.CurrentCulture, Strings.Warning_CentralPackageVersions_MultipleSourcesWithoutPackageSourceMapping, restoreRequest.Project.RestoreMetadata.Sources.Count, string.Join(", ", restoreRequest.Project.RestoreMetadata.Sources.Select(i => i.Name)))));
+                // Get a list of package sources that are not local folders
+                List<Configuration.PackageSource> nonLocalPackageSources = restoreRequest.Project.RestoreMetadata.Sources.Where(i => !i.IsLocal).ToList();
```

Comment:
We already calculated this earlier I think for telemetry purposes, so you can consider reusing it?

Reply:
I couldn't find where its calculated for telemetry, do you know where we do that?

Reply:
Nevermind, I found it:

https://github.com/NuGet/NuGet.Client/blob/b396ad8b56c089ec3fa189f1013530d237a0e8c2/src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs#L128-L131